corehq/blobs/models.py

```diff
@@ -4,3 +4,17 @@
 class BlobMigrationState(models.Model):
     slug = models.CharField(max_length=20, unique=True)
     timestamp = models.DateTimeField(auto_now=True)
+
+
+class BlobExpiration(models.Model):
+    '''
+    This models records when temporary blobs should be deleted
+    '''
+    bucket = models.CharField(max_length=255)
+    identifier = models.CharField(max_length=255)
+    expires_on = models.DateTimeField(db_index=True)
+    created_on = models.DateTimeField(auto_now=True)
+    length = models.IntegerField()
+
+    # This is set to True when the blob associated has been successfully deleted
+    deleted = models.BooleanField(default=False, db_index=True)
```

Comment:
I would hope it's pretty rare for a key to be reused since the edge cases around object existence/versioning in Riak can get a bit gnarly when reusing keys for different objects, depending on timing and what you're doing. If the object content is the same then it's not as much of an issue.